src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -137,9 +145,40 @@ static Controller fromOptions(Environment env,
         int scalingParameter = options.containsKey(STARTING_SCALING_PARAMETER) ? Integer.parseInt(options.get(STARTING_SCALING_PARAMETER)) : DEFAULT_STARTING_SCALING_PARAMETER;
         int[] scalingParameters = new int[32];
         int[] previousScalingParameters = new int[32];
-        //set the scaling parameter for each level to the starting scaling parameter or default scaling parameter
-        Arrays.fill(scalingParameters, scalingParameter);
-        Arrays.fill(previousScalingParameters, scalingParameter);
+        //check to see if we have scaling parameters stored and read those if able, otherwise use starting values
+        File f = new File(SCALING_PARAMETER_FILE_PATH, SCALING_PARAMETER_FILE_NAME);
+        if (f.length() != 0 && f.isReadable())
+        {
+            try
+            {
+                BufferedReader br = new BufferedReader(new FileReader(f));
+                String scalingParametersString = br.readLine();
+                String[] scalingParametersStrings = scalingParametersString.split(",");
+                logger.debug(String.format("Reading stored scaling parameters from %s: %s", f.toPath().toString(), Arrays.toString(scalingParametersStrings)));
+                for (int i = 0; i < scalingParameters.length; i++)
+                {
+                    //if the file does not have enough entries, use the last entry for the rest of the levels
```

Comment:
Can you add a comment explaining why this is better than using the default value for the rest of the levels?

It feels like this behaviour is a bit surprising from the user's perspective. Or is this how adaptive UCS works in other areas?

Reply:
I'm not sure if this is actually better.  When using UCS without adaptive compaction, the last element of the provided 'scaling_parameters' option is used for all higher levels, so I think it makes sense to do it the same way here.  I also can't think of an instance where this would come up since each time the scaling parameters are written to the file every level is included, but I included it just in case.